src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2002,6 +2002,20 @@ private BoundExpression BindIncrementOperator(CSharpSyntaxNode node, ExpressionS
                 hasErrors);
         }
 
+        private BoundExpression BindNotNullExpression(PostfixUnaryExpressionSyntax node, DiagnosticBag diagnostics)
+        {
+            var expr = BindValue(node.Operand, diagnostics, BindValueKind.RValue);
```

Comment:
>BindValueKind.RValue [](start = 60, length = 20)

Is this always an RValue though? Can I use ```!``` on a ```ref``` parameter, or in another "ref" context?